ESP32-C5: BLE controller asserts `adiList_stack_addAdiList:31 0 0` on extended scan (LE_Set_Extended_Scan_Enable)

## Bug description                                          
                                                                                        
On ESP32-C5, performing a BLE extended scan via LE_Set_Extended_Scan_Enable crashes   
with:                                                           
                                                                                               
`  panicked at 'ASSERT adiList_stack_addAdiList:31 0 0'         `                                  
                                                          
A legacy scan (LE_Set_Scan_Parameters / LE_Set_Scan_Enable, the pre-BT5 path) on identical
hardware and software is stable indefinitely, only the extended-scan path crashes.


## To Reproduce

1. Build the reproducer below with --release for esp32c5 and flash it to an ESP32-C5-WROOM-1
module.
2. Watch RTT - the assertion fires within seconds to a few minutes of ble scan starting.
3. Replace scanner.scan_ext(&cfg) with scanner.scan(&cfg) in the same code, then no assertion, scan
 runs just fine.

Verified to reproduce on esp-radio v0.18.0. Repro tested with a
no-op EventHandler; the assertion does not require any user-side report processing,  just
enabling extended scan in an environment with extended-adv traffic is enough.

```
#![no_std]
#![no_main]

use bt_hci::controller::ExternalController;
use embassy_executor::Spawner;
use embassy_time::{Duration, Timer};
use esp_hal::clock::CpuClock;
use esp_hal::timer::timg::TimerGroup;
use esp_radio::ble::controller::BleConnector;
use panic_rtt_target as _;
use trouble_host::prelude::*;

extern crate alloc;

esp_bootloader_esp_idf::esp_app_desc!();

const CONNECTIONS_MAX: usize = 1;
const L2CAP_CHANNELS_MAX: usize = 1;

#[esp_rtos::main]
async fn main(_spawner: Spawner) -> ! {
  rtt_target::rtt_init_defmt!();

  let peripherals = esp_hal::init(esp_hal::Config::default().with_cpu_clock(CpuClock::max()));

  esp_alloc::heap_allocator!(#[esp_hal::ram(reclaimed)] size: 65536);
  esp_alloc::heap_allocator!(size: 64 * 1024);

  let timg0 = TimerGroup::new(peripherals.TIMG0);
  let sw = esp_hal::interrupt::software::SoftwareInterruptControl::new(peripherals.SW_INTERRUPT);
  esp_rtos::start(timg0.timer0, sw.software_interrupt0);

  let transport = BleConnector::new(peripherals.BT, Default::default()).unwrap();
  let ble_controller = ExternalController::<_, 1>::new(transport);
  let mut resources: HostResources<DefaultPacketPool, CONNECTIONS_MAX, L2CAP_CHANNELS_MAX> =
      HostResources::new();
  let stack = trouble_host::new(ble_controller, &mut resources)
      .set_random_address(Address::random([0xff, 0x8f, 0x1b, 0x05, 0xe4, 0xff]));
  let host = stack.build();
  let mut runner = host.runner;
  let mut scanner = Scanner::new(host.central);

  struct H;
  impl EventHandler for H {}
  let h = H;

  let cfg = ScanConfig {
      active: false,
      phys: PhySet::M1,
      interval: Duration::from_millis(500),
      window: Duration::from_millis(100),
      timeout: Duration::from_secs(0),
      ..Default::default()
  };

  defmt::info!("ble scan starting");

  embassy_futures::join::join(
      async {
          loop {
              let _ = runner.run_with_handler(&h).await;
          }
      },
      async {
          loop {
              let _session = scanner.scan_ext(&cfg).await.unwrap();
              Timer::after(Duration::from_secs(5)).await;
          }
      },
  )
  .await;

  unreachable!()
}
```

Crashed with:
```[INFO ] ble scan starting
[ERROR] panicked at 'ASSERT adiList_stack_addAdiList:31 0 0'
[ERROR] panicked at /home/patrik/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/defmt-1.0.1/src/lib.rs:385:5:
explicit panic
```

## Things tried that did not help:
- Bumping the controller's dup lists to the max via esp_radio::ble::Config::def
ault().with_ll_adv_dup_list_count(100).with_ll_rsp_dup_list_count(100). With the default 20,
average reports until crash is ~5; with 100 it is ~30–60. Eventually crashes either way.
- Patching esp-radio os_adapter_esp32c5.rs ble_ll_scan_phy_number: 2 => 1.
- Patching esp-radio os_adapter_esp32c5.rs scan_allow_adi_filter: 0 => 1.
- Patching trouble-host host/src/scan.rs:72 FilterDuplicates::Disabled => Enabled.
- Calling set_random_address before .build().
- Short cycled sessions (e.g. 1 s scan + 500 ms idle) so the controller frees per-scan state
more often.

## Expected behavior

LE_Set_Extended_Scan_Enable on the ESP32-C5 controller should not assert and crash the firmware
 when an extended advertising frame is received. 

## Environment

- Target device: ESP32-C5-WROOM-1 (Dev-Kit)
- Crate name and version: esp-hal 1.1.0, esp-radio 0.18.0
- Other relevant deps: esp-rtos 0.3.0, bt-hci 0.8.0, trouble-host 0.6.0, embassy-executor
0.10.0, embassy-time 0.5.0

--- a/esp-radio/CHANGELOG.md
+++ b/esp-radio/CHANGELOG.md
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed a crash when using advanced BLE scanning on certain chips (C6,H2,C5,C61) (#5458)
 
 ### Removed
 

--- a/esp-radio/src/ble/npl.rs
+++ b/esp-radio/src/ble/npl.rs
@@ -1182,13 +1182,16 @@ pub(crate) fn ble_init(config: &Config) -> PhyInitGuard<'static> {
                 fn adv_stack_initEnv() -> i32;
                 fn extAdv_stack_initEnv() -> i32;
                 fn sync_stack_initEnv() -> i32;
+
+                fn base_stack_enable() -> i32;
+                fn adv_stack_enable() -> i32;
+                fn extAdv_stack_enable() -> i32;
+                fn scan_stack_enable() -> i32;
+                fn sync_stack_enable() -> i32;
             }
 
             let res = base_stack_initEnv();
             assert!(res == 0, "base_stack_initEnv returned {}", res);
-
-            let res = conn_stack_initEnv();
-            assert!(res == 0, "conn_stack_initEnv returned {}", res);
 
             let res = adv_stack_initEnv();
             assert!(res == 0, "adv_stack_initEnv returned {}", res);
@@ -1202,11 +1205,29 @@ pub(crate) fn ble_init(config: &Config) -> PhyInitGuard<'static> {
                 assert!(res == 0, "scan_stack_initEnv returned {}", res);
             }
 
+            let res = conn_stack_initEnv();
+            assert!(res == 0, "conn_stack_initEnv returned {}", res);
+
             let res = sync_stack_initEnv();
             assert!(res == 0, "sync_stack_initEnv returned {}", res);
 
             let res = r_esp_ble_msys_init(256, 320, 12, 24, 1);
             assert!(res == 0, "esp_ble_msys_init returned {}", res);
+
+            let res = base_stack_enable();
+            assert!(res == 0, "base_stack_enable returned {}", res);
+
+            let res = adv_stack_enable();
+            assert!(res == 0, "adv_stack_enable returned {}", res);
+
+            let res = extAdv_stack_enable();
+            assert!(res == 0, "extAdv_stack_enable returned {}", res);
+
+            let res = scan_stack_enable();
+            assert!(res == 0, "scan_stack_enable returned {}", res);
+
+            let res = sync_stack_enable();
+            assert!(res == 0, "sync_stack_enable returned {}", res);
         }
 
         #[cfg(feature = "coex")]
@@ -1263,6 +1284,12 @@ pub(crate) fn ble_deinit() {
         fn adv_stack_deinitEnv() -> i32;
         fn extAdv_stack_deinitEnv() -> i32;
         fn sync_stack_deinitEnv() -> i32;
+
+        fn base_stack_disable() -> i32;
+        fn adv_stack_disable() -> i32;
+        fn extAdv_stack_disable() -> i32;
+        fn scan_stack_disable() -> i32;
+        fn sync_stack_disable() -> i32;
     }
 
     #[cfg(esp32c2)]
@@ -1277,6 +1304,11 @@ pub(crate) fn ble_deinit() {
         #[cfg(not(esp32c2))]
         {
             npl::r_ble_controller_disable();
+            sync_stack_disable();
+            scan_stack_disable();
+            extAdv_stack_disable();
+            adv_stack_disable();
+            base_stack_disable();
             conn_stack_deinitEnv();
             sync_stack_deinitEnv();
             scan_stack_deinitEnv();

--- a/qa-test/Cargo.toml
+++ b/qa-test/Cargo.toml
@@ -38,6 +38,11 @@ lis3dh-async = "0.9.3"
 ssd1306 = "0.10.0"
 static_cell = "2.1.1"
 rtos-trace = { version = "0.2.0", optional = true }
+trouble-host = { version = "0.6.0", features = [
+    "default-packet-pool-mtu-255",
+    "derive",
+    "scan",
+] }
 
 [features]
 unstable = []

--- a/qa-test/Cargo.toml
+++ b/qa-test/Cargo.toml
@@ -30,7 +30,6 @@ esp-println = { path = "../esp-println", features = ["log-04"] }
 esp-radio = { path = "../esp-radio", features = [
     "log-04",
     "unstable",
-    "wifi",
 ], optional = true }
 esp-storage = { path = "../esp-storage", optional = true }
 embedded-storage = "0.3.1"

--- a/qa-test/src/bin/ext_adv_scan.rs
+++ b/qa-test/src/bin/ext_adv_scan.rs
@@ -1,0 +1,98 @@
+//! A check that extended-advertisement scanning works.
+//!
+//! To really test this you need a way to have large advertisements received.
+//!
+//! e.g. on Windows use `Bluetooth LE Explorer` - use `Advertisement Beacon`,
+//! make sure to enable "Use Extended Format" and some payload.
+//! Enable advertising.
+
+//% FEATURES: esp-radio esp-radio/ble esp-hal/unstable
+//% CHIPS: esp32 esp32s3 esp32h2 esp32c2 esp32c3 esp32c5 esp32c6 esp32c61
+
+#![no_std]
+#![no_main]
+use embassy_executor::Spawner;
+use embassy_futures::join::join;
+use embassy_time::{Duration, Timer};
+use esp_alloc as _;
+use esp_backtrace as _;
+use esp_hal::{
+    clock::CpuClock,
+    interrupt::software::SoftwareInterruptControl,
+    timer::timg::TimerGroup,
+};
+use esp_println::println;
+use esp_radio::ble::controller::BleConnector;
+use trouble_host::prelude::*;
+
+esp_bootloader_esp_idf::esp_app_desc!();
+
+#[esp_rtos::main]
+async fn main(_s: Spawner) {
+    esp_println::logger::init_logger_from_env();
+    let peripherals = esp_hal::init(esp_hal::Config::default().with_cpu_clock(CpuClock::max()));
+
+    esp_alloc::heap_allocator!(size: 72 * 1024);
+
+    let sw_int = SoftwareInterruptControl::new(peripherals.SW_INTERRUPT);
+    let timg0 = TimerGroup::new(peripherals.TIMG0);
+    esp_rtos::start(timg0.timer0, sw_int.software_interrupt0);
+
+    let bluetooth = peripherals.BT;
+    let connector = BleConnector::new(
+        bluetooth,
+        esp_radio::ble::Config::default().with_ext_adv_max_size(1650),
+    )
+    .unwrap();
+    let controller: ExternalController<_, 1> = ExternalController::new(connector);
+
+    // Using a fixed "random" address can be useful for testing. In real scenarios, one would
+    // use e.g. the MAC 6 byte array as the address (how to get that varies by the platform).
+    let address: Address = Address::random([0xff, 0x8f, 0x1b, 0x05, 0xe4, 0xff]);
+
+    println!("Our address = {:?}", address);
+
+    let mut resources: HostResources<DefaultPacketPool, CONNECTIONS_MAX, L2CAP_CHANNELS_MAX> =
+        HostResources::new();
+    let stack = trouble_host::new(controller, &mut resources).set_random_address(address);
+
+    let Host {
+        central,
+        mut runner,
+        ..
+    } = stack.build();
+
+    let printer = Printer {};
+    let mut scanner = Scanner::new(central);
+    let _ = join(runner.run_with_handler(&printer), async {
+        let config = ScanConfig {
+            active: false,
+            phys: PhySet::M1,
+            interval: Duration::from_millis(500),
+            window: Duration::from_millis(100),
+            timeout: Duration::from_secs(0),
+            ..Default::default()
+        };
+        let mut _session = scanner.scan_ext(&config).await.unwrap();
+        // Scan forever
+        loop {
+            Timer::after(Duration::from_secs(1)).await;
+        }
+    })
+    .await;
+}
+
+/// Max number of connections
+const CONNECTIONS_MAX: usize = 1;
+const L2CAP_CHANNELS_MAX: usize = 1;
+
+struct Printer {}
+
+impl EventHandler for Printer {
+    fn on_ext_adv_reports(&self, reports: LeExtAdvReportsIter<'_>) {
+        for report in reports {
+            let report = report.unwrap();
+            println!("ext adv {:?} {}", report.addr, report.data.len());
+        }
+    }
+}

--- a/qa-test/src/bin/ext_adv_scan.rs
+++ b/qa-test/src/bin/ext_adv_scan.rs
@@ -5,6 +5,8 @@
 //! e.g. on Windows use `Bluetooth LE Explorer` - use `Advertisement Beacon`,
 //! make sure to enable "Use Extended Format" and some payload.
 //! Enable advertising.
+//!
+//! Not supported on ESP32 since this feature needs BLE 5.0+ (and ESP32 is 4.2)
 
 //% FEATURES: esp-radio esp-radio/ble esp-hal/unstable
 //% CHIPS: esp32s3 esp32h2 esp32c2 esp32c3 esp32c5 esp32c6 esp32c61

--- a/qa-test/src/bin/ext_adv_scan.rs
+++ b/qa-test/src/bin/ext_adv_scan.rs
@@ -7,7 +7,7 @@
 //! Enable advertising.
 
 //% FEATURES: esp-radio esp-radio/ble esp-hal/unstable
-//% CHIPS: esp32 esp32s3 esp32h2 esp32c2 esp32c3 esp32c5 esp32c6 esp32c61
+//% CHIPS: esp32s3 esp32h2 esp32c2 esp32c3 esp32c5 esp32c6 esp32c61
 
 #![no_std]
 #![no_main]
@@ -39,11 +39,7 @@ async fn main(_s: Spawner) {
     esp_rtos::start(timg0.timer0, sw_int.software_interrupt0);
 
     let bluetooth = peripherals.BT;
-    let connector = BleConnector::new(
-        bluetooth,
-        esp_radio::ble::Config::default().with_ext_adv_max_size(1650),
-    )
-    .unwrap();
+    let connector = BleConnector::new(bluetooth, esp_radio::ble::Config::default()).unwrap();
     let controller: ExternalController<_, 1> = ExternalController::new(connector);
 
     // Using a fixed "random" address can be useful for testing. In real scenarios, one would

